src/lib/auth/README.md

```diff
@@ -0,0 +1,135 @@
+  - `createAuthClient` from `better-auth/client` is used to initialize the client-side module.
+  - The `baseURL` for the auth server is configured, defaulting to the application's domain but explicitly set for clarity and different environments (e.g., Vercel deployments).
+
+### 3. tRPC Integration
+
+- **tRPC Context (`src/server/api/trpc.ts`):**
+
+  - The `createContext` function for tRPC is augmented to include the user's session information.
+  - `auth.api.getSession()` (from the `better-auth` instance in `src/server/auth.ts`) is called on every request to retrieve the current session using request headers.
+  - The session data (including the `user` object) is then available in `ctx.session` and `ctx.user` within tRPC procedures.
+
+- **Protected Procedures (`src/server/api/trpc.ts`):**
+
+  - A `protectedProcedure` is defined, which checks for the presence of `ctx.session` and `ctx.user`. If a user is not authenticated, a `TRPCError` with code `UNAUTHORIZED` is thrown, effectively protecting backend procedures.
+
+- **Auth Router (`src/server/api/routers/auth.ts`):**
+  - This tRPC router provides specific endpoints related to authentication:
+    - `login`: Handles user login by calling `auth.api.signInEmail`.
+    - `logout`: Handles user logout by calling `auth.api.signOut`.
+    - `getSession`: A public procedure to retrieve the current session, primarily for client-side consumption.
+    - `register`: Currently throws a `NOT_IMPLEMENTED` error, as registration is intended to be handled client-side via `authClient.signUp.email`.
+  - A helper `convertToHeaders` is used to adapt Next.js request headers for `better-auth` API calls.
+
+### 4. Frontend State Management
+
+- **Auth Context (`src/contexts/AuthContext.tsx`):**
+  - An `AuthProvider` component wraps the application (or parts of it) to provide authentication state.
+  - It uses `authClient.useSession()` from `better-auth/client` to get the current session information (user, loading state).
+  - A `useAuth` hook is provided for easy access to the `user` object and `isLoading` state in any component.
+
+### 5. UI Components
+
+- **Login & Registration Forms (`src/components/auth/LoginForm.tsx`, `src/components/auth/RegisterForm.tsx`):**
+
+  - These React components provide the user interface for signing in and signing up.
+  - They use methods from the `authClient` (`authClient.signIn.email` and `authClient.signUp.email`) to interact with the `better-auth` backend.
+  - After successful operations, they typically invalidate the `auth.getSession` tRPC query to refresh the `AuthContext`.
+
+- **Auth Tile (`src/app/map/[id]/Tile/auth.dynamic.tsx`):**
+
+  - A dynamically imported component that hosts the `LoginForm` and `RegisterForm`.
+  - It allows users to switch between login and registration views.
+  - This tile is likely intended to be displayed prominently when a user is not authenticated.
+
+- **User Navigation (`src/components/layout/UserNav.tsx` - as per plan):**
```

Comment:
This component does not exist anymore for now, and it is the auth tile that will serve this purpose later on